CorsixTH/Lua/hospital.lua

```diff
@@ -1517,8 +1517,27 @@ function Hospital:initStaff()
       if added_staff then
         local staff = self.world:newEntity(profile.humanoid_class, 2)
         staff:setProfile(profile)
-        -- TODO: Make a somewhat "nicer" placing algorithm.
-        staff:setTile(self.world.map.th:getCameraTile(1))
+
+        -- Identify a safe starting place and
+        -- try to place the staff member randomly near it or on it
+        local map = self.world.map.th
+        local map_size = (map:size())
+        local x, y = map:getCameraTile(Hospital:getPlayerIndex())
+        -- Make sure the random numbers (x_attempt, y_attempt) will be safely within the map
+        local x_safe = math.max(10, math.min(x, map_size - 10))
+        local y_safe = math.max(10, math.min(y, map_size - 10))
+        local x_attempt, y_attempt
+        local attempts = 0
+        repeat
+          x_attempt = x_safe + math.random(-10, 10)
+          y_attempt = y_safe + math.random(-10, 10)
```

Comment:
Unfortunately this is not always within the map.
Your `x` and `y` are `10` or bigger from line 1527 and 1528. The reference manuals says that `math.random(m ,n)` gives a random number `[m, n]` (ie boundaries included).

The limits checks:
- Map coordinates for `x` run from `1` upto-and-including `xsize`.
- `x=10`, random = `-10`, result is `0`, which is off-map.
- `x=xsize-10`, random = `10`, result is `xsize`, which  is on-map.

These things are always tricky to get right. I made it a habit to run a check like above on any limit I set or read.

As it needs changing  anyway, perhaps add a `max_offset` variable as well, instead of the hard-coded `10` everywhere? In the current code hardcoded `10` was just still feasible, as it's `10` all over the place. But with the change, you're going to have an `11` as well somewhere, and then it becomes complicated how the `10` and `11` are related.